frontend/app/routes/public/renew/$id/ita/update-mailing-address.tsx

```diff
@@ -154,38 +113,32 @@ export async function action({ context: { appContainer, session }, params, reque
     mailingProvince: formData.get('mailingProvince') ? String(formData.get('mailingProvince')) : undefined,
     mailingCity: String(formData.get('mailingCity') ?? ''),
     mailingPostalCode: formData.get('mailingPostalCode') ? String(formData.get('mailingPostalCode')) : undefined,
-    copyMailingAddress: formData.get('copyMailingAddress') === 'copy',
-    homeAddress: formData.get('homeAddress') ? String(formData.get('homeAddress')) : undefined,
-    homeApartment: formData.get('homeApartment') ? String(formData.get('homeApartment')) : undefined,
-    homeCountry: formData.get('homeCountry') ? String(formData.get('homeCountry')) : undefined,
-    homeProvince: formData.get('homeProvince') ? String(formData.get('homeProvince')) : undefined,
-    homeCity: formData.get('homeCity') ? String(formData.get('homeCity')) : undefined,
-    homePostalCode: formData.get('homePostalCode') ? String(formData.get('homePostalCode')) : undefined,
   });
 
   if (!parsedDataResult.success) {
     return data({ errors: transformFlattenedError(parsedDataResult.error.flatten()) }, { status: 400 });
   }
 
-  const updatedData = parsedDataResult.data.copyMailingAddress
-    ? {
-        ...parsedDataResult.data,
-        homeAddress: parsedDataResult.data.mailingAddress,
-        homeApartment: parsedDataResult.data.mailingApartment,
-        homeCountry: parsedDataResult.data.mailingCountry,
-        homeProvince: parsedDataResult.data.mailingProvince,
-        homeCity: parsedDataResult.data.mailingCity,
-        homePostalCode: parsedDataResult.data.mailingPostalCode,
-      }
-    : parsedDataResult.data;
+  // If home address is the same as mailing address, populate home related properties.
+  const updatedAddressInfo = {
+    ...parsedDataResult.data,
+    ...(state.isHomeAddressSameAsMailingAddress && {
+      homeAddress: parsedDataResult.data.mailingAddress,
+      homeApartment: parsedDataResult.data.mailingApartment,
+      homeCountry: parsedDataResult.data.mailingCountry,
+      homeProvince: parsedDataResult.data.mailingProvince,
+      homeCity: parsedDataResult.data.mailingCity,
+      homePostalCode: parsedDataResult.data.mailingPostalCode,
+    }),
+  };
 
-  saveRenewState({ params, session, state: { addressInformation: updatedData } });
+  saveRenewState({ params, session, state: { addressInformation: updatedAddressInfo } });
 
   if (state.editMode) {
     return redirect(getPathById('public/renew/$id/ita/review-information', params));
   }
-
-  return redirect(getPathById('public/renew/$id/ita/dental-insurance', params));
+  // If home addres is not the same as mailing address, redirect to /update-home-address
+  return redirect(state.isHomeAddressSameAsMailingAddress ? getPathById('public/renew/$id/ita/dental-insurance', params) : getPathById('public/renew/$id/ita/update-home-address', params));
```

Comment:
Nitpick: I feel this comment can be removed because the piece of code associated with it is clear enough IMO.